File: resources/algorithmic-toolbox/week2_algorithmic_warmup/4_least_common_multiple/lcm.cpp
#include <iostream>

long long gcd(long long a, long long b) {
  if (b == 0) {
    return a;
  }

  return gcd(b, a % b);
}

long long lcm(long long a, long long b) {
  return (long long) a * b / gcd(a, b);
}

int main() {
  long long a, b;
  
  std::cin >> a >> b;
  std::cout << lcm(a, b) << std::endl;
  
  return 0;
}

File: resources/algorithmic-toolbox/week2_algorithmic_warmup/4_least_common_multiple/lcm.js
// by Alexander Nikolskiy

const readline = require('readline');

async function main (){
    const rl = readline.createInterface({
        input: process.stdin,
        terminal: false
    });
    
    process.stdin.setEncoding('utf8');

    for await (const line of rl) {
        const [a,  b] = line.split(' ');
        console.log(lcm(Number(a), Number(b)));
        process.exit();
    }
}

function gcd(a, b) {
    if (b === 0) {
        return a;
    }

    return gcd(b, a % b);
}

function lcm(a, b) {
    return a * b / gcd(a, b);
}

module.exports = lcm;

if (!module.parent) {
    main();
}